src/main/java/org/fenixedu/learning/domain/executionCourse/ExecutionCourseSite.java

```diff
@@ -56,13 +58,16 @@
     public ExecutionCourseSite(ExecutionCourse executionCourse) {
         checkNotNull(executionCourse);
         setExecutionCourse(executionCourse);
-        setPublished(true);
+
         setFolder(folderForPath(PortalConfiguration.getInstance().getMenu(), "courses"));
         setSlug(on("-").join(getExecutionCourse().getSigla(), getExecutionCourse().getExternalId()));
-        setCanAdminGroup(NobodyGroup.get());
-        setCanPostGroup(NobodyGroup.get());
-        setBennu(Bennu.getInstance());
 
+        setCreationDate(new DateTime());
+        setCanAdminGroup(DynamicGroup.get("managers"));
+        setCanPostGroup(TeacherResponsibleOfExecutionCourseGroup.get(executionCourse));
```

Comment:
I understand your comment refers to L67.
Indeed this is not the correct behavior in my view. Given the frailties of CMS and especially its lack of proper access control to support this particular use case, it is not a good idea to give PostingAccess to the teachers (responsible or not) of a certain course. Since we provide a separate mechanism to manage the CourseSite, through the Teacher portal, we should refrain from allowing the teachers to also manage the same site through the CMS portal.
That said, if this group must be initialized at all, it should be with the `NobodyGroup`.